### ISSUE TITLE
No "Save file?" dialog on closing new tab when changed but emptied

### ISSUE THREAD

**Issue by UHolder** [notepad-plus-plus/notepad-plus-plus#14674]
### Description of the Issue
If create new tab, fill with some text and then clear all text and close the tab, there will be no save file dialog before closing the tab.
So in some cases easy to lose data, because undo in this case is impossible.

### Steps to Reproduce the Issue
1. Create new tab 'New 1'.
2. Fill 'New 1' with text.
3. Use Ctrl+A (prepare to copy all text) then Del (by mistake).
4. Close 'New 1'. No Save file dialog in this case, undo of deleted text by mistake is impossible.

### Expected Behavior
Save file dialog should appear before closing new tab if tab is empty but was changed earlier.

### Actual Behavior
Save file dialog doesn't appear before closing new tab if tab is empty but was changed earlier and undo still possible.

### Debug Information
Notepad++ v8.6.2   (64-bit)
Build time : Jan 14 2024 - 02:16:00
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "readme.txt" 
Admin mode : ON
Local Conf mode : OFF
OS Name : Windows 10 Pro (64-bit)
OS Version : 22H2
OS Build : 19045.2846
Current ANSI codepage : 1251
Plugins : 
    DSpellCheck (1.5)
    HexEditor (0.9.12)
    mimeTools (3)
    NppConverter (4.5)
    NppExport (0.4)

**Comment by alankilborn** [comment]
@UHolder 

For this use case to make sense, you'd have to make TWO mistakes:  The _delete_ in step 3 plus the _close_ in step 4.
The software can't reasonably protect the user from all aspects of his own dumbness.

---

That being said, it has been requested before that an empty file should be saveable:
Steps to reproduce:
1. Ctrl+n
2. Should now be enabled but is not:  Ctrl+s
The author has declined implementing this, for reasons which are his own.
(I'm mentioning this because I was reminded of it by your issue.)

**Comment by UHolder** [comment]
This issue can definitely be classified as a Bug.
I'll explain.

First.

npp is great for various tasks.
Quick notes is one of them. The app is perfect for this.
![зображення](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/66409476/f0fbee8a-40b2-444e-81d8-61ee50d181a0)
I don't need to save all my quick notes.
During the day there is a huge flow of data, something needs to be saved for 5 seconds, something for several days.
Clear text formatting, process text with regular expressions, remember a fresh idea in one sentence, and many other examples.

And even this text I typed in npp, because if I type instantly and the browser or website or operating system crashes, I will waste time typing the text again. This comes with experience.
For task like this one npp is absolutely reliable, it will restore all notes automatically, even those that were not saved with Ctrl+S.

In high load mode when processing data anyone can easily make a mistake.
This happens to me 1-2 times a year.
Once I lost 4 hours of time in this way, it was the end of the day, I was tired, I selected the text of the quick note, cut it out using Shift+Del and automatically closed the tab. Shift wasn't pressed, npp didn't give me a chance to think I was in a hurry.

I don't want to save all my notes to disk, I don't need it, npp does it for me automatically in the background. And it's great.

Second.

Try to model the situation in MS Word. Yes, the program will also close the empty document without question, but (!) if there have been any changes before, there will definitely be a question.
This is correct editor behavior.
This comparison is made for good reason and does not require separate explanations.

**Comment by alankilborn** [comment]
>I selected the text of the quick note, cut it out using Shift+Del and automatically closed the tab. Shift wasn't pressed...

But your logic here is flawed:
Even if you were prompted to save the file in this case, you would say "no", because you think what you Cut is in the clipboard.
(A better way to do it would be to Paste the data in its destination first, then return to N++ and close out the tab)

---

But, good luck to you in getting your feature request implemented.
Perhaps the author will agree that it is a necessary thing (if so, I also suggest that he consider again to allow "new and unchanged" tabs to be saveable).

**Comment by Gilgamech** [comment]
`No save dialog when closing an empty and unsaved file.` This seems like a feature, not a bug. Why would I want to waste time being asked if I want to save an empty file? If you want to save an empty file, make an empty file and save it - that process is more "Unixy" than creating the empty file through the remains of another process. 

The premise of this Issue is flawed. It would work if the undo history were saved alongside the current data. But to my knowledge (and with a quick test), the undo history is not preserved past file close. 

You might be looking for a feature to export or save the undo history for a file, to replay that later.

**Comment by UHolder** [comment]
Nope. In this case the main is content has been modified (!). It's important,
So is no save dialog on modified file (it doesn't matter if it empty or not when closing) a feature or bug? 'Empty' - it's just file content description but not file state.

Look, empty not modified tab: 
![зображення](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/66409476/a61184e4-89f9-4da6-935a-0ed1c5eacf5b) - blue icon. Type something and empty again: 
![зображення](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/66409476/2767bea2-a905-415c-a24c-50492fe411e2) - red icon. It's doesn't stay blue again after emptying.

Even tab icon helps me to prove my position :)

**Comment by Gilgamech** [comment]
The undo history has been modified. It's probably what's driving the red icon. The content in both is "" and is probably what drives the save dialog.

**Comment by Calciferz** [comment]
> I think he has a point.

I agree, however I like closing empty "new 1" tabs without a save file dialog, so this can be argued to be a feature. 

I wonder if it would be possible to reopen the closed "new 1" tab with `Ctrl-Shift-T` instead and restore the edit history. That would also work for "new" tabs that had content and I remember making that mistake and trying to use `Ctrl-Shift-T` to no avail.
